test/extended/oauth/groupsync.go

```diff
@@ -1,111 +1,419 @@
 package oauth
 
 import (
-	"fmt"
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
 	"io/ioutil"
 	"os"
 	"os/exec"
 	"path"
+	"strings"
 
 	g "github.com/onsi/ginkgo"
 	o "github.com/onsi/gomega"
+	"gopkg.in/ldap.v2"
+	"sigs.k8s.io/yaml"
 
-	"time"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kdiff "k8s.io/apimachinery/pkg/util/diff"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/kube-openapi/pkg/util/sets"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	userv1 "github.com/openshift/api/user/v1"
+	userclient "github.com/openshift/client-go/user/clientset/versioned"
+	"github.com/openshift/library-go/pkg/security/ldapclient"
 
 	"github.com/openshift/origin/test/extended/testdata"
 	exutil "github.com/openshift/origin/test/extended/util"
-	"github.com/openshift/origin/test/extended/util/image"
-	"k8s.io/apimachinery/pkg/util/wait"
-	e2e "k8s.io/kubernetes/test/e2e/framework"
 )
 
+func init() {
+	utilruntime.Must(userv1.Install(scheme))
+}
+
 var _ = g.Describe("[sig-auth][Feature:LDAP][Serial] ldap group sync", func() {
 	defer g.GinkgoRecover()
 	var (
-		oc                 = exutil.NewCLI("ldap-group-sync")
-		remoteTmp          = "/tmp/"
-		caFileName         = "ca"
-		kubeConfigFileName = "kubeconfig"
+		oc = exutil.NewCLI("ldap-group-sync").AsAdmin()
 	)
+
 	g.It("can sync groups from ldap", func() {
 		g.By("starting an openldap server")
-		ldapService, ca, err := exutil.CreateLDAPTestServer(oc)
+		ldapNS, ldapName, _, ca, err := exutil.CreateLDAPTestServer(oc)
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		g.By("running oc adm groups sync against the ldap server")
-		_, err = oc.AsAdmin().Run("adm").Args("policy", "add-scc-to-user", "anyuid", oc.Username()).Output()
-		o.Expect(err).NotTo(o.HaveOccurred())
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel() // stop the port forward eventually
 
-		pod, err := exutil.NewPodExecutor(oc, "groupsync", image.OpenLDAPTestImage())
-		o.Expect(err).NotTo(o.HaveOccurred())
+		portFwdCmd := exec.CommandContext(ctx, "oc", "port-forward", "svc/"+ldapName, "30389:389", "-n", ldapNS)
```

Comment:
Any particular reason for manually invoking the command instead of using `oc.Run("port-forward").Args(svc/"+ldapName, "30389:389", "-n", ldapNS).Execute()`?

Reply:
I wanted to be able to gracefully terminate the command as the test finishes with the context cancellation.

Reply:
Fair.